src/lib/backup-storage-availability/backup-storage-availability.ts

```diff
@@ -0,0 +1,122 @@
+import { exec } from 'shelljs';
+import path from 'path';
+import xdgBasedir from 'xdg-basedir';
+import os from 'os';
+import checkDiskSpace from 'check-disk-space';
+import { Confirm } from 'enquirer';
+import { Job } from '../../graphqlTypes';
+
+const oneGiBInBytes = 1024 * 1024 * 1024;
+
+export class BackupStorageAvailability {
+	archiveSize: number;
+
+	constructor( archiveSize: number ) {
+		this.archiveSize = archiveSize;
+	}
+
+	static createFromDbCopyJob( job: Job ): BackupStorageAvailability {
+		const bytesWrittenMeta = job.metadata?.find( meta => meta?.name === 'bytesWritten' );
+		if ( ! bytesWrittenMeta?.value ) {
+			throw new Error( 'Meta not found' );
+		}
+
+		return new BackupStorageAvailability( Number( bytesWrittenMeta.value ) );
+	}
+
+	getDockerStorageAvailable(): number {
+		const kiBLeft = exec( `docker run --rm alpine df -k`, { silent: true } )
+			.grep( /\/dev\/vda1/ )
```

Comment:
```
$ ls -1 /dev | grep vda1
$
```

:-(

Reply:
On linux I presume? Running `/bin/sh -c 'ls -1 /dev | grep vda1'` would expectedly show this as Docker tries to expose as little as possible of the underlying system into the kernel. When I ran this inside the Docker machine `docker run -it --rm --privileged --pid=host justincormack/nsenter1`, I got a result.

That does tell me that I should pretty much skip the validation for now, if I couldn't detect a number. We should at least not crash.